server/index.js

```diff
@@ -30,6 +30,22 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:id', (req, res, next) => {
+  const { id } = req.params;
```

Comment:
Be sure to validate this. If it's not a proper integer, it's crashing your server as can be seen in your screenshot.